autobump-maven/src/main/java/model/MavenVersionRepository.java

```diff
@@ -0,0 +1,64 @@
+package model;
+
+import exceptions.DependencyParserException;
+import exceptions.WrongUrlException;
+import lombok.Getter;
+import org.apache.maven.artifact.repository.metadata.io.xpp3.MetadataXpp3Reader;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Getter
+public class MavenVersionRepository implements VersionRepository {
+    private static final String URL_SUFFIX = "maven-metadata.xml";
+    private final String baseUrl;
+
+    public MavenVersionRepository(String baseUrl) {
+        this.baseUrl = baseUrl;
+    }
+
+    @Override
+    public Set<Version> getAllAvailableVersions(Dependency dependency) {
+        try(BufferedReader in =
+                    new BufferedReader(
+                            new InputStreamReader(
+                                    getRepoUri(dependency)
+                                            .openConnection()
```

Comment:
This uses the JDK UrlConnection class. It is great to get started but it does lack some advanced configuration issues. For this PR this is fine, but I'll create a followup issue to consider using a more powerful http client here:

Examples, probably in order of preference:
- JDK 11 HttpClient (no extra dependency)
- OkHttp Client
- Apache HttpClient

Reply:
Follow up in #54